x-pack/platform/plugins/shared/cases/server/cases_analytics/tasks/backfill_task/backfill_task_runner.ts

```diff
@@ -0,0 +1,145 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger } from '@kbn/logging';
+import {
+  createTaskRunError,
+  TaskErrorSource,
+  throwRetryableError,
+  throwUnrecoverableError,
+  type ConcreteTaskInstance,
+} from '@kbn/task-manager-plugin/server';
+import type { ElasticsearchClient } from '@kbn/core/server';
+import type { CancellableTask } from '@kbn/task-manager-plugin/server/task';
+import type {
+  IndicesGetMappingResponse,
+  QueryDslQueryContainer,
+} from '@elastic/elasticsearch/lib/api/types';
+import { isRetryableEsClientError } from '../../utils';
+
+interface BackfillTaskRunnerFactoryConstructorParams {
+  taskInstance: ConcreteTaskInstance;
+  getESClient: () => Promise<ElasticsearchClient>;
+  logger: Logger;
+}
+
+export class BackfillTaskRunner implements CancellableTask {
+  private readonly sourceIndex: string;
+  private readonly destIndex: string;
+  private readonly sourceQuery: QueryDslQueryContainer;
+  private readonly getESClient: () => Promise<ElasticsearchClient>;
+  private readonly logger: Logger;
+  private readonly errorSource = TaskErrorSource.FRAMEWORK;
+
+  constructor({ taskInstance, getESClient, logger }: BackfillTaskRunnerFactoryConstructorParams) {
+    this.sourceIndex = taskInstance.params.sourceIndex;
+    this.destIndex = taskInstance.params.destIndex;
+    this.sourceQuery = taskInstance.params.sourceQuery;
+    this.getESClient = getESClient;
+    this.logger = logger;
+  }
+
+  public async run() {
+    const esClient = await this.getESClient();
+    try {
+      await this.waitForDestIndex(esClient);
+      await this.backfillReindex(esClient);
+
+      return {
+        // one time only tasks get deleted so this state is not enough
+        // for the periodic tasks to know the backfill was complete
+        state: {}, // ?
+      };
+    } catch (e) {
+      this.logger.error(`Backfill reindex of ${this.destIndex} failed. Error: ${e.message}`, {
+        tags: ['backfill-run-failed', `${this.errorSource}-error`],
+      });
+
+      if (isRetryableEsClientError(e)) {
+        throwRetryableError(
+          createTaskRunError(
+            new Error(`Backfill reindex of ${this.destIndex} failed. Error: ${e.message}`),
+            this.errorSource
+          ),
+          true
+        );
+      }
+
+      throwUnrecoverableError(createTaskRunError(e, this.errorSource));
+    }
+  }
+
+  public async cancel() {}
+
+  private async backfillReindex(esClient: ElasticsearchClient) {
+    const painlessScript = await this.getPainlessScript(esClient);
+
+    if (painlessScript.found) {
+      this.logger.info(`Reindexing from ${this.sourceIndex} to ${this.destIndex}.`);
+      await esClient.reindex({
+        source: {
+          index: this.sourceIndex,
+          query: this.sourceQuery,
+        },
+        dest: { index: this.destIndex },
+        script: painlessScript.script,
```

Comment:
There is no need to pass the whole script. The reindex operation supports passing only the painless script ID.

Reply:
I could skip the `getScript` portion that comes before this call, but isn't it better to have the confirmation that the script exists?

And once we fetch the script, we might as well send it.